File: Task/Temperature-conversion/Perl/temperature-conversion.pl
#!/usr/bin/perl
use strict ;
use warnings ;

sub kelvin_to_other_scales {
   my $kelvin = shift ;
   my $celsius = $kelvin - 273.15 ;
   my $rankine = 9 * $kelvin / 5 ;
   my $fahrenheit = $rankine - 459.67 ;
   my @scales ;
   push @scales , $kelvin , $celsius , $fahrenheit , $rankine ;
   return @scales ;
}

print "Enter a temperature in Kelvin: " ;
my $kelvin = <STDIN> ;
chomp $kelvin ;
while ( $kelvin < 0 ) {
   print "Error! There are no negative kelvin degrees!\n" ;
   $kelvin = <STDIN> ;
   chomp $kelvin ;
}
my @scales = kelvin_to_other_scales( $kelvin ) ;
print "K $scales[ 0 ]\n" ;
print "C $scales[ 1 ]\n" ;
print "F $scales[ 2 ]\n" ;
print "R $scales[ 3 ]\n" ;

File: Task/Temperature-conversion/Lua/temperature-conversion.lua
function convert_temp(k)
    local c = k - 273.15
    local r = k * 1.8
    local f = r - 459.67
    return k, c, r, f
end

print(string.format([[
Kelvin: %.2f K
Celcius: %.2f °C
Rankine: %.2f °R
Fahrenheit: %.2f °F
]],convert_temp(21.0)))
